### ISSUE TITLE
On Android, semantic widgets which are marked with link:true are announced as buttons instead of links

### ISSUE THREAD

**Issue by pcsosinski** [flutter/flutter#60039]
b/159302760

This is only observed on android, but not on iOS. When a semantic widget with link:true is focused, TalkBack announces it as a button instead of a link.

Minimal reproduction case:
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(
    MaterialApp(
      home: Scaffold(
        appBar: AppBar(title: Text('First Route')),
        body: Center(child: Semantics(link: true, child: Text('test'))),
      ),
    ),
  );
}
```
cc @gaaclarke @nscobie

**Comment by nscobie** [comment]
According to the docs for [SemanticsProperties.link](https://api.flutter.dev/flutter/semantics/SemanticsProperties/link.html), it looks like this is at least _known_ behavior:

>iOS's VoiceOver provides users with a unique hint when a link is focused. **Android's Talkback will announce a link hint the same way it does a button.**

That being said, I'm unsure if that's an Android limitation or a Flutter limitation. I'll trying poking around some more soon to see if there's anything we can improve here.

**Comment by nscobie** [comment]
From what I can tell there doesn't seem to be an easy way to handle this with the accessibility API on Android. See this [answer on StackOverflow](https://stackoverflow.com/questions/60393114/is-there-any-android-equivalent-of-ios-uiaccessibilitytraits/60406975#60406975) by "Ryan M." (Bio: _"Android @ Google.
Formerly an Android application developer, now working on androidx libraries. he/him"_) for more context.

> No, there's no concept of accessibility traits [for links] on Android - but you can still get a good accessibility experience without needing to specifically convert your layout into a Button.
> 
> Generally, it's most important that TalkBack (or whatever accessibility service is being used - remember, it's not just TalkBack) is able to detect that the widget is clickable and to be able to read a coherent description of what it does. The additional information that it's a button, specifically, isn't super useful, especially because there are so many different kinds of UI elements that it's often a very ambiguous question whether something even is a button.
> 
> You can test this by selecting it in TalkBack and confirming that it reads the content description properly, says something along the lines of "Double tap to activate," and performs the correct action when you double tap.
> 
> If it's not correct, make sure the content description, clickable flag, and click action are set correctly on the widget's AccessibilityNodeInfo.

**Comment by Piinks** [comment]
This was addressed in another issue also for `customer: money (g3)` : https://github.com/flutter/flutter/issues/36306
Link was specifically used as a button since android doesn't offer a link option in the a11y class name. 
Further reference:
- https://github.com/flutter/engine/pull/12453/files
- https://developer.android.com/reference/android/widget/Button

I think this is WAI since we intentionally did this because Android doesn't have an explicit 'link'.

**Comment by gaaclarke** [comment]
Thanks @Piinks and @nscobie !  Closing it out.
